File: Reserva-Hotel-com-Front-main/ProjetoCs-Reserva_de_Hotel-main/ReservaDeHotel/Models/ReservaHotel.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ReservaDeHotel.Models
{
    public class ReservaHotel
    {
        [Key]
        public int? IdReserva { get; set; }
        public string? NomeHospede { get; set; }
        public List<Pagamento>? Pagamento { get; set; }
        public List<EstadiaHotel>? Estadias { get; set;}

    }
}

File: Reserva-Hotel-com-Front-main/ReservaHotelFront/src/app/ReservaHotel.ts
import { Pagamento } from './Pagamento';
import { EstadiaHotel } from './EstadiaHotel';

export class ReservaHotel {
    idReserva: number = 0;
    nomeHospede: string = '';
    pagamento: Pagamento | undefined;
    estadia: EstadiaHotel | undefined;
}
